### DISCUSSION TITLE
Force terminal size columns and lines rows to exactly 80x25 or 24 for DOS CPM etc old apps

### DISCUSSION THREAD

**Discussion by john-peterson** [termux/termux-app#3578] [Ideas]
I need the best ideas to force EXACTLY 80 columns

I spent the whole day debugging a dos app that breaks if tput cols are 81. I only noticed it after I installed X11 forwarding in desperation and ran tilix through X. It opened automatically in exactly CP/M format or possibly some kind of UNIX standard format 80x24. And the text displayed fine

Anything else and the app would not print any text

They did not test this in 1981 but I didn't think of that brittleness it's the first dos app I have debugged

I only know how to resize by gesture and the closest I can get is 76 or 81 not 80

And I only know how to READ it from a typical xterm

```
tput cols
echo $COLUMNS
```

I don't know where I can set columns by default any other way. Or dynamically

**Comment by john-peterson** [comment]
For dosemu the little published setting is

```
dosemu -I 'terminal {size "80x25"}' SOFTPORN.EXE

**Reply by john-peterson** [reply]
I should have thought about this immediately

But that's the way it goes

Slow learner

**Reply by john-peterson** [reply]
And I do deal with old programs. It's not new to me

Completely unrelated

I want to create or find ANY home made program for a legendary system like December 88. Namco. Polygon machine made in Texas of all places (Houston? Nah. Impossible)

System 21 homebrew

A compiler would be incredible

I don't think there is much more than the commercial titles and the emulator

No testing programs at all have surfaced

Benchmarks

Imagine running a benchmark on System 21

3d demo

Don't know why I thought about this now but who cares

**Reply by john-peterson** [reply]
I love talking to myself

Spamming myself with unrelated comments

**Reply by john-peterson** [reply]
The title is by Al Lowe

But I don't know for certain if it's the exact replica of the first version

It's a remake in C. Quick C dialect

**Reply by john-peterson** [reply]
I even tried adding something like this myself

Before I found that it already existed for that exact reason

These programs were never tested on anything else

Reflow is unthinkable

**Reply by john-peterson** [reply]
I would not call this a high priority setting

But if someone added it to termux.properties

A startup value for the internal terminal size

I would not remove it

But any program that depend on this really should have a setting like this

So long ago terminals could not be resized

**Comment by agnostic-apollo** [comment]
You may like to spam yourself, but this is not the place for it, excessive spamming can lead to a ban.

For your problem, termux does not have any terminal width setting. Columns are automatically calculated based on font size which is changed when pinch zooming. You are not able to get exactly 80 since either your screen resolution doesn't allow that ratio or terminal horizontal margin is preventing that exact size. You need to set `terminal-margin-horizontal` to `0` or higher values and then try changing font size again so that there is enough space for 80 columns. Margin defaults to `3` and so even column numbers might be having an issue.

https://github.com/termux/termux-app/commit/63504f0adc935977b662c7257464d1fc492b6374
